pages/docs/core-concepts/error-handling.mdx

```diff
@@ -139,14 +144,70 @@ We could optionally also use the `Effect.catchTags` combinator to handle all err
 // @include: errors-1
 // ---cut---
 const program = pipe(
-//    ^?
+  //    ^?
   flakyFoo,
   Effect.zipRight(flakyBar),
   Effect.catchTags({
     FooError: (fooError) => Effect.succeed(`Recovering from ${fooError._tag}`),
-    BarError: (barError) => Effect.succeed(`Recovering from ${barError._tag}`)
+    BarError: (barError) => Effect.succeed(`Recovering from ${barError._tag}`),
   })
-)
+);
+```
+
+#### Modifying the error channel
+
+Sometimes you don't want to actually handle the error, but simplify modify it. For example, you might want to add some additional information to the error, or you might want to change the error type altogether.
+
+To do this, we can use `Effect.mapError`:
+
+```ts twoslash
+// @include: errors-1
+// ---cut---
+
+const program = pipe(
+  //    ^?
+  flakyFoo,
+  Effect.zipRight(flakyBar),
+  Effect.mapError((error) => {
+    if (error._tag === "FooError") {
+      return new Error("Something went wrong with Foo");
+    } else {
+      return new Error("Something went wrong with Bar");
+    }
+  })
+);
+```
+
+`Effect.mapError` is similar to `Effect.map`, except instead of operating on the success channel, it operates on the error channel.
+
+#### Example: Handling errors from Promises
+
+If you are using Effect with a Promise-based API, you can use the `Effect.promise` constructor to convert a Promise into an Effect. If the Promise rejects, the resulting Effect will throw with the error that the Promise rejected with. You can also use other variants such as `Effect.tryPromise` or `Effect.tryCatchPromise` to convert Promise rejections into typed errors in the Effect error channel.
```

Comment:
There are a few issues with this paragraph that I feel deserve pointing out - `promise` doesn't throw, it fails with a defect. `tryPromise` doesn't convert to a typed error, it elects failures to recoverable errors with type `unknown` (i.e. it is up to the user to refine the type). `tryCatchPromise` is the only one that elects failures to typed errors.

I would frame these three methods a bit differently:
1. `promise` - any errors thrown by the `Promise` provided to this constructor will result in a defect (i.e. an unrecoverable error)
2. `tryPromise` - any errors thrown by the `Promise` provided to this constructor will result in a recoverable error with type `unknown` in the error channel. It is then left to the user to refine the type further.
3. `tryCatchPromise` - any errors thrown by the `Promise` provided to this constructor will be provided to the `onReject` function, allowing users to elect `Promise` rejects to typed errors in the error channel

Reply:
Thanks for the feedback! Made some tweaks to reflect this information - might have simplified a bit too much initially.